src/Microsoft.Extensions.Primitives/StringTokenizer.cs

```diff
@@ -66,14 +81,8 @@ public bool MoveNext()
                     Current = default(StringSegment);
                     return false;
                 }
-
-                var next = _value.IndexOf(_separator, _index);
```

Comment:
This already exists: https://msdn.microsoft.com/en-us/library/6wh7x3fs(v=vs.110).aspx